### ISSUE TITLE
[iotile-build] Setting python_universal to false in module_settings.json causes the build to be run as if it true

### ISSUE THREAD

**Issue by x24ling** [iotile/coretools#841]
https://github.com/iotile/coretools/blob/8050a048b4048a82914f0f980e791e32dc6d0db4/iotilebuild/iotile/build/config/site_scons/pythondist.py#L176-L179

**Comment by x24ling** [comment]
Omitting 'python_universal' from module_settings.json would fix it.

Probably want to do something like
```
if tile.settings.get("python_universal", False) is False
```

**Comment by x24ling** [comment]
For reference, I ran into this issue when i tried to build con_nrf52382. 
```
<C:\Users\chris\iotile\con_btc1\con_nrf52832> iotile build
This package is set up to require py2.py3
Dependency version appears to be build\deps\iotile_standard_library_lib_controller\python\iotile_support_lib_controller_3-3.7.17-py3-none-any.whl
BuildError: dependent wheel not built with compatible python version
```

Line in module_settings.json that causes issue:
https://github.com/iotile/con_btc1/blob/master/con_nrf52832/module_settings.json#L67

**Comment by mattrunchey** [comment]
I'm confused by your suggestion. If you don't specify that it requires a universal wheel, it will require a non-universal wheel. That is because a universal wheel is more restrictive than a py2 or py3 wheel itself. 

I think the logic as it stands properly describes the requirements as per module requirements, it's the `module_settings.json` line that would need to update to reflect that it doesn't require a universal package. If con_btc1 *truly* depends on a universal package, it wouldn't be able to use coretools 5.0, since it's py3 only.

**Comment by x24ling** [comment]
Currently, the logic is that if `module_settings.json` contains the key `python_universal`, regardless if it is `True` or `False`, it will require the universal wheel. 

I suppose the logic should be
```
if tile.settings.get("python_universal", False) == False: 
     required_version = "py3" if sys.version_info[0] == 3 else "py2"     
else: 
    required_version = "py2.py3" 

```
This way if `python_universal` is either False or not defined in module_settings.json, it looks for a py2 or py3 wheel, is that correct?

**Comment by mattrunchey** [comment]
Right now, if you *dont* have the key in the module_settings.json, it will not require the universal wheel, right?

```
if "python_universal" in tile.settings:      
    required_version = "py2.py3" 
```

If the value is not present, this will evaluate to false, and it will look for py2 or py3. I believe the logic you propose is equivalent.

**Comment by x24ling** [comment]
Correct, but if the `module_settings.json` file contains the line
```
"python_universal": false,
```
It will look for a universal wheel, correct?

**Comment by mattrunchey** [comment]
Right, updated my comment to reflect that. It indeed relied on the fact that you'd either specify the `python_universal` to be true, or leave it out entirely as opposed to setting it false.

**Comment by x24ling** [comment]
That's true, nobody has used `python_universal: false` yet. 

I have labeled this issue as low priority, I don't have burning desire to fix this issue but I want to put this for reference. 

The code you wrote in this comment would work: https://github.com/iotile/coretools/issues/841#issuecomment-500635010
